pkg/apis/serving/v1/revision_defaults.go

```diff
@@ -131,7 +131,7 @@ func (rs *RevisionSpec) applyDefault(container *corev1.Container, cfg *config.Co
 		rs.applyProbes(container)
 	}
 
-	if rs.PodSpec.EnableServiceLinks == nil {
+	if rs.PodSpec.EnableServiceLinks == nil && apis.IsInCreate(ctx) {
 		rs.PodSpec.EnableServiceLinks = cfg.Defaults.EnableServiceLinks
```

Comment:
This is a noop because it's being set in the deployment?

https://github.com/knative/serving/pull/9681/files#diff-b8668ec090ebde0668133377ef4dc9f0

Reply:
But it verifies the shape is preserved correctly.

Reply:
I misread `deploy.go` and didn't see `pod.EnableServiceLinks`

```go
if cfg != nil && pod.EnableServiceLinks == nil {
    pod.EnableServiceLinks = cfg.Defaults.EnableServiceLinks
}
```